hrabsence/CRM/HRAbsence/Form/AbsenceRequest.php

```diff
@@ -679,24 +692,47 @@ private function buildSendTemplateParams() {
       'tplParams' => $tplParams,
     ];
 
-    return $sendTemplateParams;
+    return $this->setFromAddress($sendTemplateParams, $isAdmin);
+  }
+
+  /**
+   * Calculates Status for a date in a leave request.
+   * 
+   * @param int $dateValue
+   *   Either 1, 0 or null, depending on selection made by user
+   * 
+   * @return int
+   *   ID of status
+   */
+  private function calculateDateStatus($dateValue) {
+    $status = null;
+
+    if (array_key_exists('_qf_AbsenceRequest_done_save', $this->_submitValues) && isset($dateValue) && (($dateValue == 1 && $this->_showhide == 1) || ($dateValue == 0 && $this->_showhide == 0))) {
```

Comment:
please check this comment again https://github.com/civicrm/civihr/pull/1313#discussion_r93944769  .. the implementation of this method could be reduced .